.github/workflows/reusable_release_automation.yaml

```diff
@@ -0,0 +1,97 @@
+name: Release automation
+
+# This workflow encompasses two main functions:
+# 1. It checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure",
+#    and promotes it to a full release if suitable. This action should consequently trigger any workflows
+#    that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'.
+#
+# 2. It then triggers the creation of a new prerelease based on specified criteria and included files,
+#    fully automating the release cycle management.
+
+# Usage from the caller workflow:
+# jobs:
+#   release_management:
+#     uses: newrelic/coreint-automation/.github/workflows/reusable_release_automation.yaml@v3
+#     secrets: inherit
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    needs: check_block_endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.COREINT_BOT_TOKEN }}"
+        run: |
+          # Get the date of the latest release.
+          LATEST_RELEASE=$(
+            gh release list --json publishedAt,isLatest -R "${{ github.repository }}" --jq '
+              .[] | select(.isLatest) | .publishedAt | fromdateiso8601
+            '
+          )
+
+          # List all releases, filter by prereleases that was made later than the latest release, sort by date,
+          # reverse because the last one is the most recent and only grab the last prerelease
```

Comment:
instead of doing "reverse" could you use the `gh` flag?
```
 -O, --order string           Order of releases returned: {asc|desc} (default "desc")
```